wallet-new/server/Cardano/Wallet/API/V1/Swagger.hs

```diff
@@ -503,6 +583,404 @@ Disable TLS (Not Recommended)
+            "walletId": "Ae2tdPwUPE...8V3AVTnqGZ"
+        },
+        {
+            "index": 2902829384,
+            "addresses": [
+                "DdzFFzCqrh...Sj5a2rkx8y"
+            ],
+            "amount": 0,
+            "name": "MyOtherAccount",
+            "walletId": "Ae2tdPwUPE...8V3AVTnqGZ"
+        }
+    ],
+    "meta": {
+        "pagination": {
+            "totalPages": 1,
+            "page": 1,
+            "perPage": 10,
+            "totalEntries": 2
+        }
+    }
+}
+```
+
+
+Managing Addresses
+------------------
+
+By default, wallets you create are provided with an account which has one default address. It
+is possible (and recommended) for an account to manage multiple addresses. Address reuse
+actually reduces privacy for it tights more transactions to a small set of addresses.
+
+Note that when paying, this is fairly abstracted from you: addresses are selected from an
+account of your wallet based on different strategies and policies.
+
+To create a new address, use the [`POST /api/v1/addresses`](#tag/Addresses%2Fpaths%2F~1api~1v1~1addresses%2Fpost)
+endpoint:
+
+```
+curl -X POST \
+  https://localhost:8090/api/v1/addresses \
+  -H 'Content-Type: application/json;charset=utf-8' \
+  -H 'Accept: application/json;charset=utf-8' \
+  --cacert ./scripts/tls-files/ca.crt \
+  -d '{
+	"walletId": "Ae2tdPwUPE...V3AVTnqGZ4",
+	"accountIndex": 2147483648
+}'
+```
+
+```json
+{
+    "status": "success",
+    "data": {
+        "id": "DdzFFzCqrh...QfiU5E6N8S",
+        "balance": 0,
+        "used": false,
+        "changeAddress": false
+    },
+    "meta": { ... }
+}
+```
+
+If your wallet is protected with a password, this password is also required in order to create
+new addresses for that wallet. In such case, the field `spendingPassword` should match the one
+defined earlier to protect your wallet.
+
+Addresses generated as just described are always valid. However, when dealing with other
+parties, you might want to check whether a given address is a valid base58-encoded address.
+
+For that purpose, use [`GET /api/v1/addresses/{{address}}/validity`](#tag/Addresses%2Fpaths%2F~1api~1v1~1addresses~1{address}~1validity%2Fget):
+
+```
+curl -X GET https://localhost:8090/api/v1/addresses/not-a-valid-address/validity
+  -H 'Accept: application/json;charset=utf-8' \
+  --cacert ./scripts/tls-files/ca.crt
+```
+
+```json
+{
+    "status": "success",
+    "data": {
+        "valid": false
+    },
+    "meta": { ... }
+}
+```
+
+Finally, you can always view all your available addresses across all your wallets by using
+[`GET /api/v1/addresses`](#tag/Addresses%2Fpaths%2F~1api~1v1~1addresses%2Fget):
+
+```
+curl -X GET https://localhost:8090/api/v1/addresses \
+  -H 'Accept: application/json;charset=utf-8' \
+  --cacert ./scripts/tls-files/ca.crt
+```
+
+```json
+{
+    "data": [
+        "YQdiVKZwx...d16oatjqf3",
+        "2iQCuXVhU...S4YoYruxet",
+        "jM8R4tBCP...2ueD7ndDV4",
+        "B9x5woS58...TUDWnYRdfv",
```

Comment:
This isn't right either. We're returning a list of `WalletAddress` which contains more info than just the ID.